rm-datasource/src/main/java/org/apache/seata/rm/datasource/undo/AbstractUndoLogManager.java

```diff
@@ -264,9 +264,10 @@ public void undo(DataSourceProxy dataSourceProxy, String xid, long branchId) thr
             try {
                 connectionProxy = dataSourceProxy.getConnection();
                 conn = connectionProxy.getTargetConnection();
+                originalAutoCommit = conn.getAutoCommit();
 
                 // The entire undo process should run in a local transaction.
-                if (originalAutoCommit = conn.getAutoCommit()) {
+                if (conn.getAutoCommit()) {
```

Comment:
Sorry, I don't understand the purpose of doing this. Here, conn.getAutoCommit() is used twice.

Reply:
Thank you for the review. 

I intended to divdie assign to variable and check whether is true or not in If condtion. `conn.getAutoCommit()` returns only state then will be okay for using twice.

What about this alternatively?
```
originalAutoCommit = conn.getAutoCommit();
if (originalAutoCommit)
```
or
```
final boolean autoCommitState = conn.getAutoCommit();
originalAutoCommit = autoCommitState
if (autoCommitState)
```

Reply:
I prefer the first expression for its simplicity and fidelity to the original meaning.

Reply:
Thanks for your review! 
I've addressed your comments and pushed a new commit with the changes 😄